internal/store/job.go

```diff
@@ -429,5 +430,5 @@ func failureReason(jc *v1batch.JobCondition, reason string) bool {
 	if jc == nil {
 		return false
 	}
-	return jc.Reason == reason
+	return strings.EqualFold(jc.Reason, reason)
```

Comment:
We want to map the value exactly to make sure that we have a bounded list of potential values, so I don't think we should make that case insensitive. We should rather update the list of reasons and make sure we are more careful in the future to avoid these kind of typos.

Reply:
Makes sense @dgrisonnet 

I have already undo this change. Casing in https://github.com/kubernetes/kube-state-metrics/blob/main/internal/store/job.go#L42 was already part of the previous commit, so I only needed to update unit testing to make tests pass. Please let me know if something else is needed.

Thank you for the review!